metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1133,6 +1135,27 @@ def gen_dataset_workunits(
             project_id=project_id, dataset_name=dataset_name, table=table.name
         )
 
+        if (
+            isinstance(table, BigqueryTable)
+            and table.table_type is not None
+            and table.table_type == "EXTERNAL"
+            and table.ddl is not None
+            and f"CREATE EXTERNAL TABLE `{project_id}.{dataset_name}.{table.name}`"
+            in table.ddl
+        ):
+            uris_match = re.search(r"uris=\[([^\]]+)\]", table.ddl)
+            if uris_match:
+                uris_str = uris_match.group(1)
+                uris_list = json.loads(f"[{uris_str}]")
+                storage_location = str(uris_list[0])
```

Comment:
How does bigquery represent "folder of files as a dataset" scenario as mentioned [here](https://datahubproject.io/docs/generated/ingestion/sources/s3#path-specs----examples) ?

Reply:
> Is there any doc mentioning this `uris=[...]` format ? Is there only one entry in the list ?

https://cloud.google.com/bigquery/docs/external-data-cloud-storage#wildcard-support
and Yes, the uris parameter in the CREATE EXTERNAL TABLE statement in BigQuery can include multiple URIs, allowing you to reference multiple files. **All files must have the same schema.**

Reply:
> How does bigquery represent "folder of files as a dataset" scenario as mentioned [here](https://datahubproject.io/docs/generated/ingestion/sources/s3#path-specs----examples) ?
It will treat them as single table basically
This allows BigQuery to treat a collection of files stored in a GCS bucket as a single external table, which can be queried like any other table in BigQuery.

Reply:
I need to add support for multiple URIS, from UI we can attached only single uri or patterns.